nixos/modules/services/mail/sympa.nix

```diff
@@ -0,0 +1,536 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.sympa;
+  user = cfg.user;
+  group = cfg.group;
+  pkg = cfg.package.override { inherit (cfg) dataDir; };
+  dataDir = cfg.dataDir;
+  fqdns = mapAttrsToList (name: _val: name) cfg.domains;
+
+  sympaSubServices = [
+    "sympa-archive.service"
+    "sympa-bounce.service"
+    "sympa-bulk.service"
+    "sympa-task.service"
+  ];
+
+  sympaServiceConfig = srv: {
+    Type = "forking";
+    Restart = "always";
+    ExecStart = "${pkg}/bin/${srv}.pl";
+    PIDFile = "/run/sympa/${srv}.pid";
+    User = "${user}";
+    Group = "${group}";
+  };
+
+  mainConfig = pkgs.writeText "sympa.conf" ''
+    domain     ${if cfg.mainDomain != null then cfg.mainDomain else (head fqdns)}
+    listmaster ${concatStringsSep "," cfg.listMasters}
+    lang       ${cfg.lang}
+
+    home        ${dataDir}/list_data
+    arc_path    ${dataDir}/arc
+    bounce_path ${dataDir}/bounce
+
+    db_type ${cfg.database.type}
+    db_name ${cfg.database.name}
+    ${optionalString (cfg.database.host != null) "db_host ${cfg.database.host}"}
+    ${optionalString (cfg.database.port != null) "db_port ${cfg.database.port}"}
+    ${optionalString (cfg.database.user != null) "db_user ${cfg.database.user}"}
+    db_passwd #dbpass#
+
+    ${optionalString (cfg.mta.type == "postfix") ''
+      sendmail         /run/wrappers/bin/sendmail
+      sendmail_aliases ${dataDir}/sympa_transport
+
+      aliases_program ${pkgs.postfix}/bin/postmap
+      aliases_db_type hash
+    ''}
+
+    ${optionalString cfg.web.enable ''
+      static_content_path ${dataDir}/static_content
+      css_path            ${dataDir}/static_content/css
+      pictures_path       ${dataDir}/static_content/pictures
+      mhonarc             ${pkgs.perlPackages.MHonArc}/bin/mhonarc
+    ''}
+
+    ${cfg.extraConfig}
```

Comment:
Relevant is NixOS/rfcs#42

Reply:
Hmm, the config file format *seems* to be lines of `[key][whitespace][value]`, perhaps we could simply declare `settings` as `types.attrsOf types.str`. I wonder if this is good idea.

Reply:
Yeah that should work, this has many benefits as I'm describing in the RFC